dulwich/porcelain.py

```diff
@@ -1155,12 +1155,20 @@ def determine_wants(remote_refs, **kwargs):
             _import_remote_refs(r.refs, remote_name, fetch_result.refs)
 
 
-def status(repo=".", ignored=False):
+def status(repo=".", ignored=False, untracked_files="all"):
     """Returns staged, unstaged, and untracked changes relative to the HEAD.
 
     Args:
       repo: Path to repository or repository object
       ignored: Whether to include ignored files in untracked
+      untracked_files: How to handle untracked files:
+            - "no": Show no untracked directories
+                  (like `git status -uno`).
+            - "all": show all files in untracked directories
+                  (like `git status -uall`, dulwich default).
+            - "normal": shows untracked files and directories.
+                  (like `git status -unormal`, git default).
+
```

Comment:
More of a question for @jelmer, but I'm not sure whether we actually need to support `all` vs `normal` here. It could just be `untracked_files=True` instead, where true=all and false=no. 

(The same way that existing `ignored` option just supports true/false rather than the full set of git's `traditional/no/matching` modes for ignored files)

Reply:
Ideally we'd match the git command-line here. 

For convenience, we could support both strings and booleans as value for untracked_files?